pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -383,6 +387,69 @@ func ensureKubeAPIServerTrustedCA(client coreclientv1.CoreV1Interface, recorder
 	return err
 }
 
+func ensureLocalhostRecoverySAToken(client coreclientv1.CoreV1Interface, recorder events.Recorder) error {
+	requiredSA := resourceread.ReadServiceAccountV1OrDie(v410_00_assets.MustAsset("v4.1.0/kube-apiserver/localhost-recovery-sa.yaml"))
+	requiredToken := resourceread.ReadSecretV1OrDie(v410_00_assets.MustAsset("v4.1.0/kube-apiserver/localhost-recovery-token.yaml"))
+
+	saClient := client.ServiceAccounts(operatorclient.TargetNamespace)
+	serviceAccount, err := saClient.Get(requiredSA.Name, metav1.GetOptions{})
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+
+		serviceAccount, err = saClient.Create(requiredSA)
+		if err != nil {
+			return err
+		}
+	}
+
+	// The default token secrets get random names so we will create a custom secret
+	// to be populated with SA token so we have a stable name.
+	secretsClient := client.Secrets(operatorclient.TargetNamespace)
+	token, err := secretsClient.Get(requiredToken.Name, metav1.GetOptions{})
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+
+		token, err = secretsClient.Create(requiredToken)
```

Comment:
this should also be in the standard create block above.

Reply:
done

Reply:
the reconciliation logic above, where I moved it, is nuking the injected token

Reply:
It seem expected given how it is written
https://github.com/openshift/cluster-kube-apiserver-operator/blob/00bc23d7d3dcd0551c78df93249084584173ee45/vendor/github.com/openshift/library-go/pkg/operator/resource/resourceapply/core.go#L230-L239

Reply:
do we want to compare only existing keys in .Data fixture and leave the others be?

Reply:
but then you can't delete keys if you change the fixture...